src/task/src/pullRequestMetrics.ts

```diff
@@ -0,0 +1,72 @@
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import 'reflect-metadata'
+import { singleton } from 'tsyringe'
+import CodeMetricsCalculator from './metrics/codeMetricsCalculator'
+import Logger from './utilities/logger'
+import RunnerInvoker from './runners/runnerInvoker'
+
+/**
+ * A class for managing the overall PR Metrics task.
+ */
+@singleton()
+export default class PullRequestMetrics {
+  private readonly _codeMetricsCalculator: CodeMetricsCalculator
+  private readonly _logger: Logger
+  private readonly _runnerInvoker: RunnerInvoker
+
+  /**
+   * Initializes a new instance of the `PullRequestMetrics` class.
+   * @param codeMetricsCalculator The code metrics calculation logic.
+   * @param logger The logger.
+   * @param runnerInvoker The runner invoker logic.
+   */
+  public constructor (codeMetricsCalculator: CodeMetricsCalculator, logger: Logger, runnerInvoker: RunnerInvoker) {
+    this._codeMetricsCalculator = codeMetricsCalculator
+    this._logger = logger
+    this._runnerInvoker = runnerInvoker
+  }
+
+  /**
+   * Runs the overall PR Metrics task.
+   * @param folder The root folder containing index.ts.
+   * @returns A promise for awaiting the completion of the method call.
+   */
+  public async run (folder: string): Promise<void> {
+    try {
+      this._runnerInvoker.locInitialize(folder)
+
+      // TODO: Remove System.AccessToken support after a transition period has elapsed.
```

Comment:
Is there an issue logged for this?
Ideally this comment would reference the issue number.

Reply:
Makes sense. I've created #304 and updated the comment here and in another file to reference the issue number.